typescript/infra/scripts/verify-validator.ts

```diff
@@ -104,170 +89,259 @@ class S3Wrapper {
+        this.extraCheckpoints = this.extraCheckpoints.filter(
+          (j) => j < this.lastNonMissingCheckpointIndex,
         );
-        process.exit(1);
-      }),
-    ]);
+        break;
+      }
 
-  if (
-    !isLatestCheckpoint(controlLatestCheckpoint) ||
-    !isLatestCheckpoint(prospectiveLastCheckpoint)
-  )
-    process.exit(1);
-
-  console.log(`Latest Index`);
-  console.log(`control: ${controlLatestCheckpoint}`);
-  console.log(`prospective: ${prospectiveLastCheckpoint}\n`);
-
-  let extraCheckpoints = [];
-  const missingCheckpoints = [];
-  let invalidCheckpoints = [];
-  const modTimeDeltasS = [];
-  const fullyCorrectCheckpoints = [];
-  let missingInARow = 0;
-  let lastNonMissingCheckpointIndex = Infinity;
-  for (
-    let i = Math.max(controlLatestCheckpoint, prospectiveLastCheckpoint);
-    i >= 0;
-    --i
-  ) {
-    if (missingInARow == MAX_MISSING_CHECKPOINTS) {
-      missingCheckpoints.length -= MAX_MISSING_CHECKPOINTS;
-      invalidCheckpoints = invalidCheckpoints.filter(
-        (j) => j < lastNonMissingCheckpointIndex,
-      );
-      extraCheckpoints = extraCheckpoints.filter(
-        (j) => j < lastNonMissingCheckpointIndex,
-      );
-      break;
+      await this.validateCheckpointIndex(checkpointIndex);
     }
 
-    const key = `checkpoint_${i}.json`;
+    console.log(
+      `Fully correct checkpoints (${this.fullyCorrectCheckpoints.length}): ${this.fullyCorrectCheckpoints}\n`,
+    );
+    if (this.extraCheckpoints.length)
+      console.log(
+        `Extra checkpoints (${this.extraCheckpoints.length}): ${this.extraCheckpoints}\n`,
+      );
+    if (this.missingCheckpoints.length)
+      console.log(
+        `Missing checkpoints (${this.missingCheckpoints.length}): ${this.missingCheckpoints}\n`,
+      );
+    if (this.invalidCheckpoints.length)
+      console.log(
+        `Invalid checkpoints (${this.invalidCheckpoints.length}): ${this.invalidCheckpoints}\n`,
+      );
 
-    let control: Checkpoint | null;
-    let controlLastMod: Date | null;
-    try {
-      const t = await cClient.getS3Obj(key);
-      if (isCheckpoint(t.obj)) {
-        if (t.obj.checkpoint.index != i) {
-          console.log(`${i}: Control index is invalid`, t);
-          process.exit(1);
-        }
-        [control, controlLastMod] = [t.obj, t.modified];
-      } else {
-        console.log(`${i}: Invalid control checkpoint`, t);
-        process.exit(1);
-      }
-    } catch (err) {
-      control = controlLastMod = null;
+    if (this.modTimeDeltasS.length > 1) {
+      // Drop the time of the first one since it is probably way off
+      this.modTimeDeltasS.length--;
+      console.log(
+        `Time deltas (∆ < 0 -> prospective came earlier than the control)`,
+      );
+      console.log(this.modTimeDeltasS);
+      console.log(`Median: ${median(this.modTimeDeltasS)}s`);
+      console.log(`Mean:   ${mean(this.modTimeDeltasS)}s`);
+      console.log(`Stdev:  ${stdDev(this.modTimeDeltasS)}s`);
     }
```

Comment:
ditto
`modTimeDeltaS` is not assigned before this check

Reply:
See `initStatsState`.